Add separate section for Vagrant and updating the instruction

safwanrahman opened the pull request:
Update the instruction
the instruction is copied from Kuma
http://kuma.readthedocs.org/en/latest/installation-vagrant.html

r?

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+Kitsune via Vagrant
+================
+
+Core developers run Kitsune in a `Vagrant`_-managed virtual machine so we can run
```
None of the core developers use Vagrant.

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+================
+
+Core developers run Kitsune in a `Vagrant`_-managed virtual machine so we can run
+the entire Kitsune stuck.
```
Typo: stuck

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+
+Core developers run Kitsune in a `Vagrant`_-managed virtual machine so we can run
+the entire Kitsune stuck.
+If you're on Mac OS X or Linux and looking for a quick way to get started, you
```
This only talks about Mac OS X and Linux, but then has Windows instructions below. It isn't consistent.

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+       cd kitsune
+       source venv/bin/activate
+       ./manage.py createsuperuser
+   You should give the Username and Password for login into your local Kitsune.
```
There should be a blank line between this line and the code block

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+       source venv/bin/activate
+       ./manage.py createsuperuser
+   You should give the Username and Password for login into your local Kitsune.
+   Your password must contain letters and numbers
```
Nope. Passwords given during createsuperuser don't have this restriction.

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+provision the vm with `vagrant_provision in the `/bin directory
+<https://github.com/mozilla/kuma/tree/master/bin>`_.
+
+Sometimes we put vagrant declarations in the wrong order. Which means some
```
What does "Sometimes we put vagrant declarations in the wrong order." mean? Can you clarify this? Wouldn't it be better to eliminate any problems with the Vagrant declarations, instead of this work-around?

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+On Ubuntu, ``vagrant up`` might fail after being unable to mount NFS shared
+folders. First, make sure you have the nfs-common and nfs-server packages
+installed and also note that you can't export anything via NFS inside an
+encrypted volume or home dir.
```
I'm pretty sure this part about encrypted drives and encrypted homedirs is just wrong, considering I use an encrypted drive, and have used Vagrant. Do you have any evidence to back up this claim?

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,119 @@
+Note: If you decide to run ``nfs: false``, the system will be a lot slower.
+There is also the potential of running into weird issues with puppet,
+since the current puppet configurations do not currently support
+``nfs: false``.
```
We don't use puppet at all.

mythmon commented:
There is a lot of stuff here that is inconsistent, not relevant to Kitsune, or just wrong. Please fact check.

safwanrahman commented:
@mythmon, I have corrected is as per needed! I have removed the Error section as I dont know whats the common Error after fixing the Vagrant. If any problem comes in future, we will figure out and write in the docs

mythmon commented on docs/installation-vagrant.rst:
```diff
@@ -0,0 +1,63 @@
+
+   At the end, you should see::
+
+      => default: notice: Finished catalog run in .... seconds
```
This is a Puppet message. It won't be shown on kitsune.

mythmon commented:
Besides still referencing Puppet indirectly, the new docs look better to me. I'd be fine merging these.

@willkg what do you think of these?

mythmon commented:
I talked with Will. He doesn't have much of an opinion here. I say this is ok to merge after we fix the problem with the "At the end you should see..." part.

safwanrahman commented:
@mythmon I have fixed that and squashed it into one commit. I hope its ready for merge.

But, till the vagrant don't get fixed, I think `vagrant up` will not work. Therefore I hope that will be fixed soon.

safwanrahman commented:
@mythmon Should it be merged?
